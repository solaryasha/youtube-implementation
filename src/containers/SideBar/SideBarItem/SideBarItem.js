import React from 'react';
import { Menu, Icon } from 'semantic-ui-react';
import './SideBarItem.scss';

const SideBarItem = (props) => {
  const highlight = props.highlight ? 'highlight-item' : null;

  return (
    <Menu.Item className={['sidebar-item', highlight].join(' ')}>
      <div className="sidebar-item-alignment">
        <span><Icon name={props.name} size="large" /></span>
        <span>{props.label}</span>
      </div>
    </Menu.Item>
  );
};

export default SideBarItem;
