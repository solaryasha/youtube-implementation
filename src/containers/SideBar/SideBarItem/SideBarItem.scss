.sidebar-item {
    
    span {
        i.icon {
            margin-right: 20px;
            color:#888888;
        }
    }

    &.highlight-item {
        span {
            font-weight: 600;

            i.icon {
                color:#ff0002;
            }
        }
    }
}

.sidebar-item-alignment {
    display: flex;
    align-items: center;
}