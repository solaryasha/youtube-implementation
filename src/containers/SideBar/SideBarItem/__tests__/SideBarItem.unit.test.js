import React from 'react';
import {shallow} from 'enzyme';
import SideBarItem from '../SideBarItem';

test('SideBar item rendering', () => {
    const wrapper = shallow(
        <SideBarItem />
    );

    expect(wrapper).toMatchSnapshot();
})