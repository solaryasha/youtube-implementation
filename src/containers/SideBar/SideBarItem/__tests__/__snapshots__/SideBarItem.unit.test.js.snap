// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar item rendering 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;
